ts.js

```diff
@@ -124,7 +125,28 @@
          }
       ],
       '@typescript-eslint/no-inferrable-types': ['error', {'ignoreParameters': true}],
-      '@typescript-eslint/no-misused-promises': ['error', {'checksConditionals': true, 'checksVoidReturn': false}]
+      '@typescript-eslint/no-misused-promises': ['error', {'checksConditionals': true, 'checksVoidReturn': false}],
+      '@typescript-eslint/class-literal-property-style': 'error',
+      '@typescript-eslint/no-confusing-non-null-assertion': 'error',
+      '@typescript-eslint/no-extraneous-class': 'error',
+      '@typescript-eslint/no-meaningless-void-operator': 'error',
+      '@typescript-eslint/no-non-null-asserted-nullish-coalescing': 'error',
+      'no-throw-literal': 'off',
+      '@typescript-eslint/no-throw-literal': 'error',
+      '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
+      '@typescript-eslint/no-unnecessary-condition': 'error',
+      'no-useless-constructor': 'off',
+      '@typescript-eslint/no-useless-constructor': 'error',
+      '@typescript-eslint/non-nullable-type-assertion-style': 'error',
+      '@typescript-eslint/prefer-for-of': 'error',
+      '@typescript-eslint/prefer-function-type': 'error',
+      '@typescript-eslint/prefer-includes': 'error',
+      '@typescript-eslint/prefer-nullish-coalescing': 'error',
+      '@typescript-eslint/prefer-optional-chain': 'error',
+      '@typescript-eslint/prefer-reduce-type-parameter': 'error',
+      '@typescript-eslint/prefer-string-starts-ends-with': 'error',
+      '@typescript-eslint/prefer-ts-expect-error': 'error',
+      '@typescript-eslint/no-non-null-assertion': 'off'
```

Comment:
what is this detail set to `off`?

Reply:
somewhere along the way, one of the rulesets we're inheriting enabled this rule, and when I was working through it, I didn't like it. It forbids you from using `obj!` to cast obj to the non-null version of its type, and while I don't think this should be done willy-nilly, there are plenty of valid cases to do this.

Reply:
yes, i like the `!` syntax, there are many cases where you _know_ it will not be null and i don't want unnecessary conditionals either.  the same thing exists now in C#.

ty.